src/state/actions/annotation.js

```diff
@@ -50,6 +52,105 @@ export function receiveAnnotationFailure(targetId, annotationId, error) {
   };
 }
 
+/**
+ * coerceAnnotationToCanvasId - hack to force `on` attribute to canvas ID
+ *
+ * @param  {String} targetId
+ * @param  {Object} annotationJson
+ * TODO: also handle v3 target attribute
+ */
+function coerceAnnotationToCanvasId(targetId, annotationJson) {
+  const coercedJson = Object.assign({}, annotationJson);
+
+  coercedJson.resources = annotationJson.resources.map((resource) => {
+    const coercedResource = Object.assign({}, resource);
+    coercedResource.on = [].concat(coercedResource.on);
+    if (coercedResource.on[0].includes('xywh=')) {
+      coercedResource.on[0] = coercedResource.on[0].replace(/^[^#]+/, targetId); // replace up to hash
+    }
+    return coercedResource;
+  });
+
+  return coercedJson;
+}
+
+/**
+ * filterAnnotationResources - filter annotation resources
+ *
+ * @param  {Object} annotationJson
+ * TODO: make this into a configuration option
+ */
+function filterAnnotationResources(annotationJson) {
+  const filteredJson = Object.assign({}, annotationJson);
+
+  filteredJson.resources = annotationJson.resources.filter(resource => resource.dcType === 'Line');
+
+  return filteredJson;
+}
+
+/**
+ * fetchAnnotationResourcesFulltext - fetch annotation fulltext
+ *
+ * @param  {Object} annotationJson
+ */
+function fetchAnnotationResourcesFulltext(annotationJson) {
+  const urls = annotationJson.resources
+    .filter(resource => !resource.resource.chars && resource.resource['@id'])
+    .map(resource => resource.resource['@id'].split('#')[0]);
+
+  const fulltext = {};
+
+  // TODO: error handling
+  const fetches = uniq(urls).map(url => axios.get(url)
```

Comment:
Is the reason for using axios because it converts the response to JSON automatically? to avoid adding dependencies could we not just use built-in fetch and add `.then(response => response.json())` before line 105?
https://stackoverflow.com/a/42493030/2474466 

or is our JSON not always UTF-encoded?

Reply:
axios was used because it automagically handles the encoding of the characters in various languages, whereas unfetch, and indeed the native fetch method in a browser console, result in mis-encoded characters that are then output as other symbols in the annotations panel. Whether that indicates a problem at source with the encoding of the annotations in the API response, I am uncertain, as noted in the ticket description, but have kept the axios workaround here in the meantime.

Reply:
got it, ok